### PR TITLE
Fix #5103 - ElectricEquipmentITEAirCooled / ElectricEquipmentITEAirCooledDefinition - align objects with other SpaceLoads

### LINKED ISSUE #5103: Deprecate ElectricEquipmentITEAirCooledDefinition's "Watts per Zone Floor Area" and replace with "Watts per Space Floor Area"
Enhancement Request
--------------

Deprecate ElectricEquipmentITEAirCooledDefinition's "Watts per Zone Floor Area" and replace with "Watts per Space Floor Area"

## Detailed Description

cf https://github.com/NREL/OpenStudio/issues/5090#issuecomment-1988079322

## Possible Implementation

Straightforward.

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #5103 
 - All changes are perfectly backward compatible, but I did deprecate 10 methods

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: resources/model/OpenStudio.idd [hunk 1]
```diff
@@ -4140,7 +4140,7 @@ OS:ElectricEquipment:ITE:AirCooled:Definition,
        \units W
        \ip-units W
        \minimum 0
-  N2 , \field Watts per Zone Floor Area
+  N2 , \field Watts per Space Floor Area
```
Comment (jmarrec) [inline resources/model/OpenStudio.idd:4143]: IDD change

File: src/model/ElectricEquipmentITEAirCooled.hpp [hunk 2]
```diff
@@ -107,13 +109,21 @@ namespace model {
...
 
-    boost::optional<double> wattsperUnit() const;
+    /** Returns the designLevel if possible based on the underlying data of electricEquipmentITEAirCooledDefinition(). */
+    boost::optional<double> designLevel() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperUnit() const;
 
-    boost::optional<double> wattsperZoneFloorArea() const;
+    /** Returns the designLevel if possible based on the underlying data of electricEquipmentITEAirCooledDefinition(). */
+    boost::optional<double> powerPerFloorArea() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperZoneFloorArea() const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooled.hpp:118]: This is definitely the case for all SpaceLoadInstance objects

File: src/model/ElectricEquipmentITEAirCooled.hpp [hunk 3]
```diff
@@ -107,13 +109,21 @@ namespace model {
     /** @name Other */
     //@{
 
-    boost::optional<double> wattsperUnit() const;
+    /** Returns the designLevel if possible based on the underlying data of electricEquipmentITEAirCooledDefinition(). */
+    boost::optional<double> designLevel() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperUnit() const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooled.hpp:114]: That one is debatable I suppose. Lights for eg also doesn't use `designLevel` but `lightingLevel`. But I think Lights is wrong, and all SpaceLoadInstance should follow this convention.

File: src/model/ElectricEquipmentITEAirCooledDefinition.hpp [hunk 4]
```diff
@@ -63,9 +65,11 @@ namespace model {
 
     std::string designPowerInputCalculationMethod() const;
 
-    boost::optional<double> wattsperUnit() const;
+    boost::optional<double> designLevel() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperUnit() const;
 
-    boost::optional<double> wattsperZoneFloorArea() const;
+    boost::optional<double> wattsperSpaceFloorArea() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperZoneFloorArea() const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooledDefinition.hpp:72]: This one is a sure thing. Per Zone Floor Area makes no sense.

File: src/model/ElectricEquipmentITEAirCooledDefinition.hpp [hunk 5]
```diff
@@ -63,9 +65,11 @@ namespace model {
 
     std::string designPowerInputCalculationMethod() const;
 
-    boost::optional<double> wattsperUnit() const;
+    boost::optional<double> designLevel() const;
+    OS_DEPRECATED(3, 8, 0) boost::optional<double> wattsperUnit() const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooledDefinition.hpp:69]: Also debatable I suppose, but I stand by it.

File: src/model/ElectricEquipmentITEAirCooledDefinition.hpp [hunk 6]
```diff
@@ -189,9 +195,17 @@ namespace model {
...
 
-    double getWattsperUnit(double floorArea) const;
+    /** Returns the design level represented by this definition, assuming floorArea (m^2). */
+    double getDesignLevel(double floorArea) const;
+    OS_DEPRECATED(3, 8, 0) double getWattsperUnit(double floorArea) const;
 
-    double getWattsperZoneFloorArea(double floorArea) const;
+    /** Returns the watts/m^2 represented by this definition, assuming floorArea (m^2) */
+    double getPowerPerFloorArea(double floorArea) const;
+    OS_DEPRECATED(3, 8, 0) double getWattsperZoneFloorArea(double floorArea) const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooledDefinition.hpp:204]: :heavy_check_mark:

File: src/model/ElectricEquipmentITEAirCooledDefinition.hpp [hunk 7]
```diff
@@ -189,9 +195,17 @@ namespace model {
     /** @name Other */
     //@{
 
-    double getWattsperUnit(double floorArea) const;
+    /** Returns the design level represented by this definition, assuming floorArea (m^2). */
+    double getDesignLevel(double floorArea) const;
+    OS_DEPRECATED(3, 8, 0) double getWattsperUnit(double floorArea) const;
```
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooledDefinition.hpp:200]: Debatable but I prefer it!

File: developer/doc/ReleaseNotes/OpenStudio_Release_Notes_3_8_0_TBD.md [hunk 8]
```diff
@@ -75,13 +75,21 @@ You can also refer to the [OpenStudio SDK Python Binding Version Compatibility M
...
+    * The constructor will no longer assign an effectiveness at 75% and assume a constant effectiveness. A helper method `bool assignHistoricalEffectivenessCurves()` is provided to create and assign 4 `TableLookup`s that will match the pre E+ 24.1.0 defaults for Sensible/Latent Effectiveness at 75% Heating/Cooling airflow
+* [#5105](https://github.com/NREL/OpenStudio/pull/5105) - ElectricEquipmentITEAirCooled / ElectricEquipmentITEAirCooledDefinition - align objects with other SpaceLoads
+    * A number of methods have been renamed (and the old ones deprecated) to conform to the API for other `SpaceLoadInstance` / `SpaceLoadDefinition` objects
+    * Mostly `getWattsperUnit` is changed to `getDesignLevel` and `getWattsperZoneFloorArea` is changed to `getPowerPerFloorArea`
+    * Refer to [deprecated_methods.csv](../../ruby/deprecated_methods.csv) for the complete list
```
Comment (jmarrec) [inline developer/doc/ReleaseNotes/OpenStudio_Release_Notes_3_8_0_TBD.md:88]: * Release notes, including for #5099

File: developer/ruby/deprecated_methods.csv
Comment (jmarrec) [inline developer/ruby/deprecated_methods.csv:1]: Updated the deprecated_methods.csv for this PR + any missing ones before

File: src/model/ElectricEquipmentITEAirCooled.hpp [hunk 9]
```diff
@@ -107,13 +109,21 @@ namespace model {
...
 
-    boost::optional<double> wattsperZoneFloorArea() const;
+    /** Returns the designLevel if possible based on the underlying data of electricEquipmentITEAirCooledDefinition(). */
```
Comment (joseph-robertson) [inline src/model/ElectricEquipmentITEAirCooled.hpp:116]: "Returns the powerPerFloorArea if possible ..."?

**Review by joseph-robertson — COMMENTED** [review]
Just one minor comment, otherwise LGTM.

File: src/model/ElectricEquipmentITEAirCooled.hpp (hunk 9 above)
Comment (jmarrec) [inline src/model/ElectricEquipmentITEAirCooled.hpp:116]:
```suggestion
    /** Returns the powerPerFloorArea (Design Level) if possible based on the underlying data of electricEquipmentITEAirCooledDefinition(). */
```

**Pull request merged**